app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -241,25 +245,38 @@ def build_metric_id(type, resource, metric_id)
     # by ems_ref, which in Hawkular terms is the
     # fully qualified resource path from Hawkular inventory
     def run_generic_operation(operation, ems_ref, parameters = {})
-      with_provider_connection do |connection|
         the_operation = {
           :operationName => operation,
           :resourcePath  => ems_ref.to_s,
           :parameters    => parameters
         }
+        run_operation(the_operation)
+    end
 
+    def run_specific_operation(operation_name, ems_ref, parameters = {})
+      parameters[:resourcePath] = ems_ref.to_s
+      run_operation(parameters, operation_name)
+    end
+
+    def run_operation(parameters, operation_name = nil)
+      with_provider_connection do |connection|
         actual_data = {}
-        connection.operations(true).invoke_generic_operation(the_operation) do |on|
+        callback = proc do |on|
           on.success do |data|
             _log.debug "Success on websocket-operation #{data}"
             actual_data[:data] = data
           end
           on.failure do |error|
-            actual_data[:data]  = {}
             actual_data[:error] = error
             _log.error 'error callback was called, reason: ' + error.to_s
```

Comment:
I guess we can leave it as is for now. And when MiQ has the notification center revisit to make the message more clear.
Also I wonder if we should feed that info success/error into the the MiQ event system to show up in the timeline view. This is a bit of a grey area, as for many operations the immediate reponse is something like
```
:data=>{"message"=>"The request has been forwarded to feed [829c6e5f-402f-4c34-b202-676631735c12] (MessageId=ID:65a957dc-54d6-11e6-8599-21d16e823d31)"}}
```
which does not make too much sense to the user.